core/src/main/java/lucee/runtime/functions/arrays/ArrayToStruct.java

```diff
@@ -35,20 +35,24 @@
 
 	private static final long serialVersionUID = 2050803318757965798L;
 
-	public static Struct call(PageContext pc, Array arr) throws PageException {
-		Struct sct = new StructImpl();
+	public static Struct call(PageContext pc, Array arr, boolean valueAsKey) throws PageException {
+		Struct sct = new StructImpl(Struct.TYPE_LINKED);
 		int[] keys = arr.intKeys();
```

Comment:
when touching that code, i would also improve it, instead of getting the key array and then get every single value separately, we should do here `arr.getEntryIterator()` and then do the following loop below
Entry a;
while(it.hasNext()) {
   e=it.next();

Reply:
done